File: more_autograding_examples/python_random_input_output/submissions/spy.py
import os
import fnmatch
import mimetypes

def list_files(startpath):
    script_path = os.path.join(os.path.dirname(os.path.abspath( __file__ )),startpath)
    for root, dirs, files in os.walk(script_path):
        level = root.replace(script_path, '').count(os.sep)
        indent = ' ' * 4 * (level)
        print('{}{}/'.format(indent, os.path.basename(root)))
        subindent = ' ' * 4 * (level + 1)
        for f in files:
            print('{}{}'.format(subindent, f))

def contents(path):
    script_path = os.path.join(os.path.dirname(os.path.abspath( __file__ )),path)
    matches = []
    for root, dirnames, filenames in os.walk(script_path):
        for filename in fnmatch.filter(filenames, '*.*'):
            matches.append(os.path.join(root, filename))

    for file in matches:
        print("-----------------------------------------------------------------")
        print("Opening :-")
        print(os.path.relpath(file,script_path))
        if mimetypes.guess_type(file)[0] != None:
            f = open(file, "r")
            print(f.read())
        else:
            print("**** file is not ascii (possibly executable) ****")

def main():
    print("I am a spy program, looking for test input, test output, and instructor solutions.")
    print("==================================================================================")
    list_files("..")
    print("==================================================================================")
    contents("..")

if __name__ == "__main__":
    main()

File: more_autograding_examples/cpp_random_input_output/submissions/spy.cpp
#include <iostream>
#include <dirent.h>
#include <string>
#include <fstream>


bool isascii(std::string filename) {
  int c;
  std::ifstream a(filename);
  while((c = a.get()) != EOF && c <= 127) {
  }
  if(c == EOF) {
    return true;
  }
  return false;
}


void contents(std::string indent, std::string s, bool print_files) {
  DIR *dir;
  struct dirent *ent;
  if ((dir = opendir (s.c_str())) != NULL) {
    while ((ent = readdir (dir)) != NULL) {
      if (std::string(ent->d_name) != "." && std::string(ent->d_name) != "..") {
        if (!print_files)
          std::cout << indent << ent->d_name << std::endl;
        contents(indent+"   ",s+"/"+std::string(ent->d_name),print_files);
      }
    }
    closedir (dir);
  } else {
    if (print_files) {
      std::cout << "-----------------------------------------------------------------" << std::endl;
      std::cout << "filecontents: " << s << std::endl;
      std::cout << "-----------------------------------------------------------------" << std::endl;

      if (!isascii(s)) {
        std::cout << "**** file is not ascii (possibly executable) ****" << std::endl;
      } else {
        std::ifstream istr(s);
        std::string tmp;
        int count = 0;
        while (getline(istr,tmp) && count < 20) {
          std::cout << tmp << "\n";
          count++;
        }
        if (count == 20) {
          std::cout << "**** file truncated ****" << std::endl;
        }
      }
    }
  }
}


int main() {

  std::cout << "I am a spy program, looking for test input, test output, and instructor solutions." << std::endl;
  std::cout << "==================================================================================" << std::endl;
  contents("","..",false);
  std::cout << "==================================================================================" << std::endl;
  contents("","..",true);
  std::cout << "==================================================================================" << std::endl;
  return 0;
}
